lib/engines/content_object_store/test/unit/app/services/update_edition_service_test.rb

```diff
@@ -0,0 +1,274 @@
+
+    @original_content_block_edition = build(:content_block_edition, details: { "foo" => "Foo text", "bar" => "Bar text" })
+    @original_content_block_edition.document = content_block_document
+    @original_content_block_edition.save!
+  end
+
+  describe "#call" do
+    let(:content_id) { "49453854-d8fd-41da-ad4c-f99dbac601c3" }
+    let(:schema) { build(:content_block_schema, block_type: "content_block_type", body: { "properties" => { "foo" => "", "bar" => "" } }) }
+    let(:edition_params) do
+      {
+        content_block_document_attributes: {
+          title: "Some Title",
+          block_type: "email_address",
+        }.with_indifferent_access,
+        details: {
+          "foo" => "Foo text",
+          "bar" => "Bar text",
+        },
+        creator: build(:user),
+      }
+    end
+
+    setup do
+      # This UUID is created by the database so instead of loading the record
+      # we stub the initial creation so we know what UUID to check for.
+      ContentObjectStore::ContentBlockEdition.any_instance.stubs(:create_random_id)
+                                             .returns(content_id)
+
+      ContentObjectStore::ContentBlockSchema.stubs(:find_by_block_type)
+                                            .returns(schema)
+    end
+
+    test "it returns the new content block edition so the controller can redirect" do
+      result = ContentObjectStore::UpdateEditionService.new(schema, @original_content_block_edition)
+                                                       .call(edition_params)
+      assert_instance_of ContentObjectStore::ContentBlockEdition, result
+    end
+
+    test "it does not create a new ContentBlockDocument" do
+      original_count = ContentObjectStore::ContentBlockDocument.count
+      ContentObjectStore::UpdateEditionService.new(schema, @original_content_block_edition)
+                                              .call(edition_params)
+      assert_equal original_count, ContentObjectStore::ContentBlockDocument.count
+    end
+
+    test "updates the title field on the original ContentBlockDocument" do
+      result = ContentObjectStore::UpdateEditionService.new(schema, @original_content_block_edition)
+                                                       .call(edition_params)
+      assert_equal result.document.title, edition_params[:content_block_document_attributes][:title]
+    end
+
+    describe "when a document title isn't provided" do
+      test "does not update the document" do
+        edition_params.delete(:content_block_document_attributes)
+
+        assert_no_changes -> { @original_content_block_edition.document.title } do
+          ContentObjectStore::UpdateEditionService
+            .new(schema, @original_content_block_edition)
+            .call(edition_params)
+        end
+      end
+    end
+
+    describe "when no parameters are changed" do
+      it "publishes a new edition with the same values as the original" do
```

Comment:
This edge case is something we might want to handle differently in future. For the user to make a change but submit the same results is a redundant action, we could accept the request, recognise no change would be made and prevent any changes, returning success like a 304 Not Modified. That would be a bit more involved so for now I think we just trust the user not to click submit when they've not actually made any changes, since that would make them the author... perhaps that's a actually workaround for someone who wants to take on ownership.